Add extended test for Request Headers IdP

stlaz opened the pull request.

enj commented on test/extended/oauth/requestheaders.go:
```diff
@@ -0,0 +1,287 @@
+}
+
+func getOAuthServerHost(routeClient routev1client.Interface) string {
+	oauthRoute, err := routeClient.RouteV1().Routes("openshift-authentication").Get("oauth-openshift", metav1.GetOptions{})
```
use well-known

stlaz replied:
done

enj commented on test/extended/oauth/requestheaders.go:
```diff
@@ -0,0 +1,390 @@
+				name:        "/authorize: valid cert",
+				cert:        goodCert,
+				key:         goodKey,
+				endpoint:    "/oauth/authorize?client_id=openshift-challenging-client&response_type=token",
```
You need at least some tests for a browser based client id, i.e. not `openshift-challenging-client`

stlaz replied:
Added two + /oauth/token/request, a bit too simple, but it's something

stlaz commented on test/extended/oauth/requestheaders.go:
```diff
@@ -0,0 +1,390 @@
+
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true, // we don't care about the server certs here
```
I'm not sure the router CA appears in the system bundle but it should not be hard to retrieve it, I'll add that, perhaps there could be cases where even that would fail.

stlaz commented on test/extended/oauth/requestheaders.go:
```diff
@@ -0,0 +1,390 @@
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
```
neat, thanks

stlaz commented:
Addressed most comments, still to-do:
- remove TLS server cert skip
- add common names to config, add a test-case for that
- add tests for `openshift-browser-client`

stlaz commented:
/test e2e-aws-serial

stlaz commented:
Added some fairly simple browser-client tests

stlaz commented on test/extended/oauth/requestheaders.go:
```diff
@@ -0,0 +1,390 @@
+		changeTime := time.Now()
+		oauthCluster, err := oc.AdminConfigClient().ConfigV1().OAuths().Get("cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		oauthCluster.Spec.IdentityProviders = []configv1.IdentityProvider{
```
I added `clientCommonNames`, but I guess the idea is to have two different providers and try to play with this a bit? Sounds like a reasonable follow-up.

stlaz commented:
/test e2e-aws-upgrade
/test verify

stlaz commented:
/test e2e-aws-serial
let's see whether it's green by random or not

enj commented:
/lgtm
/refresh
